stdlib/string.mc

```diff
@@ -235,6 +235,40 @@ utest strTrim " aaaa   " with "aaaa"
 utest strTrim "   bbbbb  bbb " with "bbbbb  bbb"
 utest strTrim "ccccc c\t   \n" with "ccccc c"
 
+-- `strEndsWith str suffix` returns true if `str` ends with the provided `suffix`
+let strEndsWith : String -> String -> Bool
+  = lam str. lam suffix. 
+    let strLen = length str in 
+    let suffixLen = length suffix in 
+    if lti strLen suffixLen then
+      false
+    else
+      eqString (subsequence str (subi strLen suffixLen) strLen) suffix
```

Comment:
These functions are reimplementations of `isSuffix` and `isPrefix` from `seq.mc`. I think it makes sense to have string specific versions (and keep the tests!), but they should be defined as
```
let strEndsWith : String -> String -> Bool = isSuffix eqChar

let strStartsWith : String -> String -> Bool = isPrefix eqChar
```

Reply:
Done! Although I did change it to `lam s1. lam s2. isSuffix eqChar s2 s1` because I feel that order of arguments makes more sense.

Reply:
I would say that the regular order makes more sense due to that it plays nicer with partial application. As an example, let's say that I want to find all MCore files in a directory:
```
let mcore_files = filter (strEndsWith ".mc") (sysListDir ".") in
```

Reply:
Good point. I've changed it.